demisto_sdk/commands/download/downloader.py

```diff
@@ -1,741 +1,978 @@
+            content_item_type=content_item_type
+        )
+        downloaded_items: list[dict]
+
+        if content_item_type == ContentItemType.AUTOMATION:
+            downloaded_items = self.get_system_automation(
+                content_items=self.input_files
+            )
+
+        elif content_item_type == ContentItemType.PLAYBOOK:
+            downloaded_items = self.get_system_playbook(content_items=self.input_files)
 
-            if self.system_item_type == "Automation":
-                system_items_list = self.get_system_automation(req_type)
+        else:
+            logger.info(
+                f"Fetching system items from server ({self.client.api_client.configuration.host})..."
+            )
+            api_response = demisto_client.generic_request_func(
+                self.client,
+                endpoint,
+                req_type,
+                body=req_body,
+                response_type="object",
+            )[0]
+
+            if content_item_type in (
+                ContentItemType.CLASSIFIER,
+                ContentItemType.MAPPER,
+            ):
+                if classifiers_data := api_response.get("classifiers"):
+                    downloaded_items = classifiers_data
 
-            elif self.system_item_type == "Playbook":
-                system_items_list = self.get_system_playbook(req_type)
+                else:
+                    logger.debug(
+                        "Could not find expected 'classifiers' key in API response.\n"
+                        f"API response:\n{json.dumps(api_response)}"
+                    )
+                    downloaded_items = []
 
             else:
-                api_response = demisto_client.generic_request_func(
-                    self.client, endpoint, req_type, body=req_body
+                downloaded_items = api_response
+                logger.info(
+                    f"Successfully downloaded {len(downloaded_items)} system content items."
                 )
-                system_items_list = ast.literal_eval(api_response[0])
 
-            system_items_list = self.arrange_response(system_items_list)
+        content_objects: dict[str, dict] = {}
 
-            for item in system_items_list:  # type: ignore
-                file_name: str = self.build_file_name(item)
-                file_path: str = os.path.join(self.system_content_temp_dir, file_name)
-                write_dict(file_path, data=item)
-
-            return True
+        for content_item in downloaded_items:
+            file_name = self.generate_content_file_name(
+                content_item=content_item, content_item_type=content_item_type
+            )
+            file_data = StringIO(json.dumps(content_item))
+            content_object = self.create_content_item_object(
+                file_name=file_name, file_data=file_data, _loaded_data=content_item
+            )
+            content_objects[file_name] = content_object
```

Comment:
```suggestion
            file_names_to_content_objects[file_name] = content_object
```

Reply:
Seems like an unnecessarily long name to me?
There's also a docstring explaining the structure of this dict on the function that creates it.

File names are already stored within the "objects" that are dicts containing information about the custom content (ID, name, filename, file data, etc.).
I used a dict instead of a list to be able to access specific content items without having to go on a loop over a list of dicts to search and find a specific one each time.

There are many dicts in use here, and we don't name them by a 'key_exaplanation_to_value_explanation' convention, so I don't fully understand why we need to here.